NIFI-14556 - normalizing JSON git-registered flows

pvillard31 opened the pull request:
# Summary

[NIFI-14556](https://issues.apache.org/jira/browse/NIFI-14556) - normalizing JSON git-registered flows

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

exceptionfactory reviewed (changes requested):
Thanks for working on this improvement @pvillard31, having better normalization of the JSON is a definite improvement for Git diff of flow definitions.

I raised a couple comments on the approach, but one general question, did you evaluate options on ObjectMapper, or module-based implementation? Jackson has a wide variety of configurable features, I mentioned one of them, although sorting objects based on the `identifier` field would still require custom handling. Either way, I think it would be worth some closer evaluation of implementation options.

pvillard31 commented on nifi-extension-bundles/nifi-extension-utils/nifi-git-flow-registry/src/main/java/org/apache/nifi/registry/flow/git/serialize/JacksonFlowSnapshotSerializer.java:
```diff
@@ -19,40 +19,98 @@
+
+            if (!elems.isEmpty()) {
+                JsonNode first = elems.get(0);
+                if (first.isObject() && first.has("identifier")) {
```
Added some comments

pvillard31 commented:
Thanks for the review @exceptionfactory - I have addressed your comments in this PR. In addition, I also implemented an option that is leveraging the custom module option. This alternative is in this PR: https://github.com/apache/nifi/pull/9948
